youtube_dl/extractor/bilibili.py

```diff
@@ -227,6 +242,15 @@ def _real_extract(self, url):
             ('<h1[^>]+\btitle=(["\'])(?P<title>(?:(?!\1).)+)\1',
              '(?s)<h1[^>]*>(?P<title>.+?)</h1>'), webpage, 'title',
             group='title') + ('_p' + str(page_id) if page_id is not None else '')
+
+        # Add the part value in for anthologies
+        if page_id is not None:
+            anthology_link = "https://api.bilibili.com/x/player/pagelist?bvid=%s&jsonp=jsonp" % bv_id
+            json_str = self._download_webpage(anthology_link, video_id,
+                                          note='Extracting videos in anthology')
+            parsed_json = json.loads(json_str)
```

Comment:
Since this same code is running for the anthology playlist itself, it would be best if it didn't have to run again for each video